From ec1cb27a9f231268064260b5ae12660582eeb4be Mon Sep 17 00:00:00 2001
From: Don Zickus <dzickus@redhat.com>
Date: Tue, 16 Sep 2014 17:04:40 -0400
Subject: [usb] usbtest: fix the bit mask of usb 2.0 extension descriptor

Message-id: <1410887463-20674-209-git-send-email-dzickus@redhat.com>
Patchwork-id: 95639
O-Subject: [RHEL7 PATCH 208/591] usb: usbtest: fix the bit mask of usb 2.0 extension descriptor
Bugzilla: 1110939
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>

https://bugzilla.redhat.com/show_bug.cgi?id=1110939

commit 875bc23ac041519b24bd2c1faf71822e4b3e4322
Author: Huang Rui <ray.huang@amd.com>
Date:   Wed Nov 13 22:35:14 2013 +0800

    usb: usbtest: fix the bit mask of usb 2.0 extension descriptor

    USB 2.1 Link PM adds to use bits[1:15] according to USB 2.0 ECN Errata for
    Link Power Management spec.

    Bit Encoding
    0 Reserved
    1 LPM
    2 BESL & Altemate HIRD definitions supported
    3 Recommended Baseline BESL valid
    4 Recommended Deep BESL valid
    11:8 Recommended Baseline BESL value
    15:12 Recommended Deep BESL value
    31:16 Reserved

    So fix the bit mask from 0x1e to 0xfffe.

    Reported-by: Sarah Sharp <sarah.a.sharp@linux.intel.com>
    Signed-off-by: Huang Rui <ray.huang@amd.com>
    Acked-by: Felipe Balbi <balbi@ti.com>
    Acked-by: Sarah Sharp <sarah.a.sharp@linux.intel.com>
    Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/usb/misc/usbtest.c b/drivers/usb/misc/usbtest.c
index f3c3136..0b5c3b1 100644
--- a/drivers/usb/misc/usbtest.c
+++ b/drivers/usb/misc/usbtest.c
@@ -619,8 +619,8 @@ static int is_good_ext(struct usbtest_dev *tdev, u8 *buf)
  }
 
  attr = le32_to_cpu(ext->bmAttributes);
- /* bits[1:4] is used and others are reserved */
- if (attr & ~0x1e) { /* reserved == 0 */
+ /* bits[1:15] is used and others are reserved */
+ if (attr & ~0xfffe) { /* reserved == 0 */
   ERROR(tdev, "reserved bits set\n");
   return 0;
  }
-- 
1.7.1